.github/workflows/testing.yml

```
@@ -9,50 +9,42 @@ on:
 
 env:
   FORCE_COLOR: 2
-
-permissions:
-  contents: read
+  coverage_node: 18
+  coverage_os: ubuntu-latest
 
 jobs:
   lint:
     uses: stylelint/.github/.github/workflows/lint.yml@main
 
   test:
-    name: Test on Node.js ${{ matrix.node }} and ${{ matrix.os }}
-    runs-on: ${{ matrix.os }}
-
-    strategy:
-      fail-fast: false
-      matrix:
-        node: [14, 16, 18]
-        os: [ubuntu-latest, windows-latest, macos-latest]
-
+    uses: stylelint/.github/.github/workflows/test.yml@main
+    with:
+      os: '["ubuntu-latest", "windows-latest", "macos-latest"]'
+      exclude: '[{"node-version": "${{ env.coverage_node }}", "os": "${{ env.coverage_os }}"}]'
+
+  test-coverage:
+    name: Test on Node.js ${{ env.coverage_node }} and ${{ env.coverage_os }} for coverage
+    runs-on: ${{ env.coverage_os }}
     steps:
-      - uses: actions/checkout@v3
+      - name: Checkout
+        uses: actions/checkout@v3
 
       - name: Set up Node.js
         uses: actions/setup-node@v3
         with:
-          node-version: ${{ matrix.node }}
+          node-version: ${{ env.coverage_node }}
           cache: npm
 
       - name: Install latest npm
-        # TODO: npm@9 is the latest, but it doesn't work on Node.js v12 and v14 with windows.
-        run: npm install --global npm@8
+        run: npm install --global npm
 
       - name: Install dependencies
         run: npm ci
 
-      - name: Test
-        run: npm run jest
-        if: "!(startsWith(matrix.os, 'ubuntu') && matrix.node == 18)"
-
       - name: Test with coverage
-        run: npm run jest -- --coverage
-        if: startsWith(matrix.os, 'ubuntu') && matrix.node == 18
+        run: npm run test:coverage
```

```suggestion
        run: npm run test-coverage
```

Minor nit as we use `:` to signify a subtask that can be run in parallel, e.g. `lint:*`.

---

Make sense. Fixed via 3827f4be.